src/lib.rs

```diff
@@ -473,13 +503,15 @@ fn link_is_publishable(link: &PackageLink) -> bool {
 
 /// Is a particular package publishable.
 ///
-/// A package is publishable if either publication is unrestricted or the one
-/// and only registry it is allowed to be published to is "crates.io".
-fn package_is_publishable(pkg: &PackageMetadata) -> bool {
+/// A package is publishable if either publication is unrestricted or it can be
+/// published to one registry.
+fn package_is_publishable(pkg: &PackageMetadata, registry: Option<&str>) -> bool {
+    let registry_target = registry.unwrap_or("crates.io");
+
     let result = match pkg.publish() {
         guppy::graph::PackagePublish::Unrestricted => true,
         guppy::graph::PackagePublish::Registries(registries) => {
-            registries.len() == 1 && registries[0] == "crates.io"
+            registries.len() == 1 && registries[0] == registry_target
```

Comment:
To facilitate publishing to private registries, I settled on using the passed argument as check. This defaults to `crates.io` if undefined.